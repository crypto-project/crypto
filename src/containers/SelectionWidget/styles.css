.selection-widget {
  margin-top: 100px;
}

.selection-widget .ui.segment {
  height: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
}
