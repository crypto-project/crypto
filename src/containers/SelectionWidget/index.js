// @flow

import React from "react";
import { Segment } from "semantic-ui-react";
import "./styles.css";

type Props = {|
  price: number,
  volume: number,
  change: number,
  base: string,
  target: string
|};

const SelectionWidget = ({ price, volume, change, base, target }: Props) => {
  return (
    <div className="selection-widget">
      <Segment>
        <div className="currency-data">
          <div className="x-price">{price}</div>
          <div className="x-volume">{volume}</div>
          <div className="x-change">{change}</div>
          <div className="x-base">{base}</div>
          <div className="x-target">{target}</div>
        </div>
      </Segment>
    </div>
  );
};

export default SelectionWidget;
