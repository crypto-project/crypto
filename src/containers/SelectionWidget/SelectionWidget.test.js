import React from "react";
import { shallow, mount } from "enzyme";
import SelectionWidget from "containers/SelectionWidget";
import { Segment } from "semantic-ui-react";

describe("I want to see currency data within the selection widget", () => {
  it("renders a panel", () => {
    const wrapper = shallow(<SelectionWidget />);

    expect(wrapper.find(Segment).exists()).toBe(true);
  });

  // data = base, target, price, volume, change
  it("renders exchange rate data ", () => {
    const wrapper = shallow(
      <SelectionWidget
        base="BTC"
        target="USD"
        price={4352.58}
        volume={57218.36}
        change={83.72}
      />
    );
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".x-price").text()).toMatch(wrapper.props().price);
    expect(wrapper.find(".x-volume").text()).toMatch(wrapper.props().volume);
    expect(wrapper.find(".x-change").text()).toMatch(wrapper.props().change);
    expect(wrapper.find(".x-base").text()).toMatch(wrapper.props().base);
    expect(wrapper.find(".x-target").text()).toMatch(wrapper.props().target);
  });
});
