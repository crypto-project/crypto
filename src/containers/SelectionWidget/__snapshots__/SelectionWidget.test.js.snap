// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I want to see currency data within the selection widget renders exchange rate data  1`] = `
<div
  className="selection-widget"
>
  <Segment>
    <div
      className="currency-data"
    >
      <h1>
        hello world
      </h1>
    </div>
  </Segment>
</div>
`;
